### DISCUSSION TITLE
Moving footer to inside each doc

### DISCUSSION THREAD

**Discussion by xixixao** [facebook/docusaurus#8771] [Q&A] [answered]
Currently footer renders in `Layout` at the bottom of the page. We want to push the footer inside each individual doc, so that the sidebar extends all the way to the bottom of the body. Is it possible to achieve with some swizzling?

**Comment by slorber** [comment, marked as answer]
Pseudo-code idea to try:


```jsx
import React from 'react';

import Footer from '@theme-original/Footer';
import {useLocation} from '@docusaurus/router';


export default function FooterWrapper(props): JSX.Element | null {
  const {pathname} = useLocation();
  const shouldRender = props.forceRender || !pathname.startsWith("/docs/")
  if (!shouldRender)  {
    return null;
  }
  return (
    <Footer  {...props} />
  );
}
```

By default `<Footer>` will not render for `/docs/*` pages 

Then use `<Footer forceRender/>` inside the docs layout component?

**Reply by xixixao** [reply]
Since we only have docs it turned out to be pretty easy with only wrapping (and some CSS):

wrap DocItem/Layout:

```jsx
import Layout from "@theme-original/DocItem/Layout";
import Footer from "@theme-original/Footer";

export default function LayoutWrapper(props) {
  return (
    <>
      <Layout {...props} />
      <Footer />
    </>
  );
}
```

wrap Footer:

```jsx
export default function FooterWrapper(props) {
  return null;
}
```

I think my struggle was due to not realizing that "DocItem/Layout" was the component I needed to swizzle.
